mkall/MKReporter.h

```diff
@@ -0,0 +1,63 @@
+// Part of Measurement Kit <https://measurement-kit.github.io/>.
+// Measurement Kit is free software under the BSD license. See AUTHORS
+// and LICENSE for more information on the copying conditions.
+#ifndef MKReporter_h
+#define MKReporter_h
+
+#import <Foundation/Foundation.h>
+
+/// MKReporterSettings contains the settings of MKReporterTask.
+@interface MKReporterSettings : NSObject
+
+/// init initializes the settings with default values.
+-(instancetype)init;
+
+/// setBaseURL sets the optional collector base URL. If no base URL is
+/// provided, we will try to discover a base URL using the bouncer.
+-(void)setBaseURL:(NSString *)baseURL;
+
+/// setTimeout sets the optional per-request timeout. By default we
+/// don't set any timeout for submitting measurements.
+-(void)setTimeout:(int64_t)timeout;
```

Comment:
Because the timeout is a per-request property, it's unfortunate to have it here. It would be much more convenient if the timeout was instead passed to `MKReporterTask:submit`.